app/models/agents/twitter_user_agent.rb

```diff
@@ -57,13 +51,15 @@ def default_options
         'username' => 'tectonic',
         'include_retweets' => 'true',
         'exclude_replies' => 'false',
-        'expected_update_period_in_days' => '2'
+        'expected_update_period_in_days' => '2',
+        'choose_home_time_line' => 'true'
       }
     end
 
     def validate_options
-      errors.add(:base, "username is required") unless options['username'].present?
       errors.add(:base, "expected_update_period_in_days is required") unless options['expected_update_period_in_days'].present?
+      
+      errors.add(:base, "username is required") unless options['choose_home_time_line'] == 'true'
```

Comment:
This will cause the validation to fail even if a `username` is provided, i think you want 
```
errors.add(:base, "username is required") if options['username'].blank? && !boolify(options['choose_home_time_line'])
```